geopandas/geoseries.py

```diff
@@ -795,6 +806,16 @@ def explode(self):
         GeoDataFrame.explode
 
         """
+        if add_multiindex is None:
```

Comment:
We should probably only raise this warning if `ignore_index` is not True? (because then the `add_multiindex` keyword is ignored anyway)

(and same for geodataframe)

Reply:
Yes, that a good catch.

Reply:
So the tests currently only test `df.explode(ignore_index=True, add_multiindex=True)`, but so where doing `add_multiindex=True` or `False` doesn't matter, as it is ignored. But if you don't specify it, you get the warning.

Reply:
I added a commit to try to address this (please take a look at it)